wandb/interface/interface.py

```diff
@@ -255,9 +255,13 @@ def _make_exit(self, exit_code):
     def _make_config(self, config_dict, obj=None):
         config = obj or wandb_internal_pb2.ConfigData()
         for k, v in six.iteritems(config_dict):
-            update = config.update.add()
-            update.key = k
-            update.value_json = json.dumps(v)
+            try:
+                value_json = json.dumps(v)
+                update = config.update.add()
+                update.key = k
+                update.value_json = value_json
+            except TypeError:
```

Comment:
We have this in cli-og:  https://github.com/wandb/client/blob/master/wandb/util.py#L480  We try to convert common types like numpy into serializable json.  @raubitsj do we have something similar in cling?

Reply:
Yes,  But i didnt apply it to config.   In the same file interface/interface.py look at send_history() and do something similar.

Reply:
@vanpelt So, I tried running the value through the `_safer` implementation, and I get the same error I did before running against huggingface (`TypeError: Object of type 'device' is not JSON serializable`). It's an error I don't get on cli-og, so I'm thinking there has to be something else in that code path. Do you know where in cli-og the config gets serialized and synced?

Reply:
https://github.com/wandb/client/blob/master/wandb/wandb_config.py#L268